main.go

```diff
@@ -120,15 +124,20 @@ func generate(in configInput) ([]byte, error) {
 				},
 			},
 		},
+		// We set the projectNumber and networkName in both the `id` and `metadata`
+		// fields. xDS v2 implementation in TD expects these in the latter while the
+		// v3 implementation expects these in the former. Once we stop supporting
+		// v2, we can get rid of these metadata entries.
 		Node: &node{
-			Id:      uuid.New().String() + "~" + in.ip,
+			Id:      fmt.Sprintf(nodeIDFormatStr, in.gcpProjectNumber, in.vpcNetworkName, uuid.New().String()),
 			Cluster: "cluster", // unused by TD
 			Locality: &locality{
 				Zone: in.zone,
 			},
 			Metadata: map[string]string{
 				"TRAFFICDIRECTOR_NETWORK_NAME":       in.vpcNetworkName,
 				"TRAFFICDIRECTOR_GCP_PROJECT_NUMBER": strconv.FormatInt(in.gcpProjectNumber, 10),
+				"INSTANCE_IP":                        in.ip,
```

Comment:
Where did this key name come from?

Reply:
This is mentioned in the Security Alpha User Guide.